assets/javascripts/kitten/components/form/autocomplete/index.js

```diff
@@ -0,0 +1,293 @@
+    color: ${COLORS.font2};
+  }
+
+  :focus {
+    border-color: ${COLORS.line2};
+  }
+
+  ::-ms-clear {
+    display: none;
+  }
+
+  ${({ error }) =>
+    error &&
+    css`
+      border-color: ${COLORS.error3};
+      color: ${COLORS.error3};
+
+      :focus {
+        border-color: ${COLORS.line2};
+        color: ${COLORS.font1};
+      }
+    `}
+`
+
+const Suggestions = styled.ul`
+  position: absolute;
+  top: ${pxToRem(50)};
+  left: 0;
+  right: 0;
+  overflow-y: auto;
+  margin: 0;
+  padding: 0;
+
+  background: ${COLORS.background1};
+  border: ${pxToRem(2)} solid ${COLORS.line1};
+  border-top: none;
+
+  list-style: none;
+
+  ${({ itemsLength }) =>
+    itemsLength > 0 &&
+    css`
+      height: ${pxToRem(
+        itemHeight * (itemsLength > 2 ? maxItemsVisibled : itemsLength),
+      )};
+    `}
+`
+
+const Item = styled.li`
+  padding: ${pxToRem(10)} ${pxToRem(15)};
+
+  ${TYPOGRAPHY.fontStyles.light};
+  font-size: 14px;
+  line-height: 1.3;
+  color: ${COLORS.font1};
+
+  cursor: pointer;
+  transition: background-color 0.2s;
+
+  :hover,
+  :focus,
+  :active {
+    background-color: ${COLORS.background3};
+  }
+
+  :focus {
+    outline: none;
+  }
+
+  &[aria-selected='true'] {
+    background-color: ${COLORS.line1};
+  }
+`
+
+export const Autocomplete = ({
+  items: defaultItems,
+  error,
+  onChange,
+  onBlur,
+  onKeyDown,
+  ...props
+}) => {
+  const [items, setItems] = useState(defaultItems)
+  const [value, setValue] = useState(props.defaultValue)
+  const [selectedItemIndex, setSelectedItemIndex] = useState(-1)
+  const [showSuggestions, setShowSuggestions] = useState(false)
+  const inputEl = useRef(null)
+  const suggestionsEl = useRef(null)
+
+  useEffect(() => {
+    updateSuggestions()
+    setShowSuggestions(!!value)
+  }, [value])
+
+  const handleChange = e => {
+    setValue(e.target.value)
+
+    onChange()
+  }
+
+  const handleBlur = () => {
+    setTimeout(() => {
+      setShowSuggestions(false)
+    }, 100)
+
+    onBlur()
+  }
+
+  const handleKeyDown = e => {
+    const enterKeyCode = 13
+    const escKeyCode = 27
+    const arrowUpKeyCode = 38
+    const arrowDownKeyCode = 40
+
+    if (e.keyCode === escKeyCode) setShowSuggestions(false)
+
+    if (showSuggestions) {
+      if (e.keyCode === arrowUpKeyCode) {
+        e.preventDefault()
+        prevSelectedItem()
+      }
+
+      if (e.keyCode === arrowDownKeyCode) {
+        e.preventDefault()
+        nextSelectedItem()
+      }
+
+      if (e.keyCode === enterKeyCode) {
+        e.preventDefault()
+
+        const selectedValue = items[selectedItemIndex]
+        handleClickItem(selectedValue)()
+      }
+    }
+
+    onKeyDown()
+  }
+
+  const handleClickItem = value => () => {
+    if (!value) return
+
+    inputEl.current.value = value
+    inputEl.current.focus()
+
+    setValue(value)
+    setShowSuggestions(false)
+  }
+
+  const updateSuggestions = () => {
+    const regex = new RegExp(`^${value}`, 'gi')
```

Comment:
Je serai d'avis d'être plus flexible sur la recherche, et de chercher le mot, peu importe l'endroit où il se trouve :
```suggestion
    const regex = new RegExp(`.*${value}.*`, 'gi')
```

Reply:
J'ai l'impression que si c'est une expression régulière il faudrait échapper la valeur, car on ne veut pas que les caractères `\ ^ $ * + ? . ( ) | { } [ ]` dans `value` aient un sens et cassent l'expression.

Une autre stratégie serait de ne pas utiliser une regex ici mais de juste vérifier que le text est contenu dedans (avec `.includes()` ou `indexOf()`).

Reply:
Du coup il faudrait modifier la casse du résultat et de la recherche via `toLowerCase()` par précaution.